### PR TITLE
Modify how vars are parsed to allow for additional fields

### PR DESCRIPTION
**Pull request by adjackura**
Vars can now be marked as required and have a description while still allowing for the old var style:
"Vars": {
  "some-normal-var": "some-string"
  "some-required-var": {"Required": true, "Description": "this is a description"}
}

Fix logging for subworkflows

### CODE REVIEW

File: daisy/workflow/workflow.go [hunk 1]
```diff
@@ -223,20 +238,52 @@ func (w *Workflow) populateStep(step *Step) error {
...
+		// Don't overwrite existing vars (applies to subworkflows).
+		if _, ok := w.vars[k]; ok {
+			continue
+		}
+		var sv string
+		if err := json.Unmarshal(v, &sv); err == nil {
+			w.vars[k] = vars{Value: sv}
+			continue
+		}
+		var vv vars
```
Comment (crunk1) [inline daisy/workflow/workflow.go]: Nit: in practice, we've been doing "vs" to denote plural.

File: daisy/workflow/workflow.go [hunk 2]
```diff
@@ -223,20 +238,52 @@ func (w *Workflow) populateStep(step *Step) error {
...
+			}
+			w.vars[k] = vv
+			continue
+		}
+		return fmt.Errorf("cannot unmarshal Var %q, value: %s", k, v)
+	}
+	return nil
+}
+
 func (w *Workflow) populate() error {
```
Comment (crunk1) [inline daisy/workflow/workflow.go]: I've been trying to parse the changes here. Is anything going to change with how autovars vs vars get resolved?

File: daisy/workflow/workflow.go [hunk 3]
```diff
@@ -247,17 +294,12 @@ func (w *Workflow) populate() error {
 		"USERNAME":  w.username,
 	}
 
-	vars := map[string]string{}
-	for k, v := range w.Vars {
-		vars[k] = v
-	}
-
 	var replacements []string
```
Comment (crunk1) [inline daisy/workflow/workflow.go]: why not map[string]string?

File: daisy/workflow/workflow.go (hunk 1 above)
Comment (adjackura) [inline daisy/workflow/workflow.go]: This isn't plural, well 'vars' is but that may be a bad struct name. sv above was string value, this was vars value

File: daisy/workflow/workflow.go (hunk 3 above)
Comment (adjackura) [inline daisy/workflow/workflow.go]: not sure what that block was here at all, i just use w.vars directly

File: daisy/workflow/workflow.go (hunk 2 above)
Comment (adjackura) [inline daisy/workflow/workflow.go]: No, at least there shouldn't be.

File: daisy/workflow/workflow.go (hunk 3 above)
Comment (crunk1) [inline daisy/workflow/workflow.go]: No, I mean why isn't replacements map[string]string?

Comment (crunk1) [inline daisy/workflow/workflow.go]: Oh, nvm.

**Pull request merged**